zerver/lib/send_email.py

```diff
@@ -149,8 +150,15 @@ def render_templates() -> Tuple[str, str, str]:
     if from_address == FromAddress.support_placeholder:
         from_address = FromAddress.SUPPORT
 
-    # Set the "From" that is displayed separately from the envelope-from
-    extra_headers["From"] = str(Address(display_name=from_name, addr_spec=from_address))
+    # Set the "From" that is displayed separately from the envelope-from.
+    # Check that the sender address is not overly-long for SES compatibility.
+    # If the address is still too long without the name an SMTPDataError will
+    # be raised when sending the email and it will be caught to raise an
+    # EmailNotDeliveredException
```

Comment:
This is a bit longer than we really need -- I think we can just say "Amazon SES rejects emails with From names longer than 320 characters; since it's better to send an email without a name than not send it."

Reply:
It's not just SES; [as noted in the issue](https://github.com/zulip/zulip/issues/17558#issuecomment-801245790), this is actually an RFC limitation.